rdr_service/dao/study_nph_dao.py

```diff
@@ -1397,3 +1398,56 @@ def update(self, obj: Incident) -> Incident:
         obj.message = truncated_value
         with self.session() as session:
             return self.update_with_session(session, obj)
+
+
+class DlwDosageDao(BaseDao):
+
+    def __init__(self):
+        super(DlwDosageDao, self).__init__(DlwDosage)
+        self.participant_dao = NphParticipantDao()
+
+    def get_id(self, obj):
+        pass
+
+    def _validate(self, nph_participant_id, resource):
+        valid_modules = [str(m) for m in list(ModuleTypes.numbers())]
+        valid_visit_periods = list(VisitPeriod.names())
+
+        # Ensure nph participant id exists
+        with self.session() as session:
+            self.participant_dao.get_id(session=session, nph_participant_id=nph_participant_id)
+
+        # Ensure all values are provided
+        for key, val in resource.items():
+            if not val:
+                raise BadRequest(f'{key} is a required field.')
+
+        # Validate Module
+        module = resource.get('module')
+        if module not in valid_modules:
+            raise BadRequest(f'Invalid module provided. Module should be in {valid_modules}')
+
+        # Validate Visit Period
+        visit_period = resource.get('visitperiod', '').upper()
+        if visit_period not in valid_visit_periods:
+            raise BadRequest(f'Invalid visitPeriod provided. Visit Period should be in {valid_visit_periods}')
+
+    # pylint: disable=unused-argument
+    def from_client_json(self, resource, participant_id, client_id):
```

Comment:
Can we try to pass this as an underscore so we don't have to use the pylint bypass?

`from_client_json(self, resource, participant_id, _):`

Reply:
@dtharpeuno I think we do need to pass in the `client_id` explicitly here b/c of this
```
def _get_model_to_insert(self, resource, participant_id=None):
        # Children of participants accept a participant_id parameter to from_client_json; others don't.
        if participant_id:
            return self.dao.from_client_json(
                resource, participant_id=participant_id, client_id=app_util.get_oauth_id()
            )
        else:
            return self.dao.from_client_json(resource, client_id=app_util.get_oauth_id())
```